crates/storage/src/storage.rs

```diff
@@ -23,6 +23,8 @@ impl Default for Storage {
 
 #[async_trait]
 impl StorageProvider for Storage {
```

Comment:
Can we add config to actually use the new S3 provider & wire it up? I think that will reveal some more problems.